### PR TITLE
ASIM Network Session schema parser with its sample and test data for SentinelOne

### PR DESCRIPTION
**Pull request by jayeshprajapaticrest**
Required items, please complete
   
   Change(s):
   - Added ASIM Network Session schema parser for SentinelOne based on API Data

   Reason for Change(s):
   - Initial version ASIM Network Session Parser

   Version Updated:
   - Initial version

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-sudkharat** [comment]
Hello @jayeshprajapaticrest, thank you for raising Pull Request with us! We will review it internally and get back to you by shortly.

**Comment by vakohl** [comment]
ETA for review is 18th Aug

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
+Parser:
+  Title: Network Session ASIM filtering parser for SentinelOne
+  Version: '0.1.1'
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: SentinelOne Documentation
+- Link: https://<SOneInstanceDomain>.sentinelone.net/api-doc/overview
+Description: |
+  This ASIM parser supports normalizing SentinelOne logs to the ASIM Network Session normalized schema. SentinelOne events are captured through SentinelOne data connector which ingests SentinelOne server objects such as Threats, Agents, Applications, Activities, Policies, Groups, and more events into Microsoft Sentinel through the REST API.
+ParserName: ASimNetworkSessionSentinelOne
+EquivalentBuiltInParser: _Im_NetworkSession_SentinelOne
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          EventSeverity = iff(EventOriginalSeverity == "Critical", "High", EventOriginalSeverity),
+          SrcDvcIdType = iff(isnotempty(DvcId), "Other", ""),
+          DvcIdType = iff(isnotempty(DvcId), "Other", ""),
+          SrcUsernameType = _ASIM_GetUsernameType(SrcUsername)
+      | extend
+          Dvc = coalesce(DvcId, DvcHostname, DvcIpAddr)
+      | extend
+          EventCount = int(1),
+          EventProduct = "SentinelOne",
+          EventResult = "Success",
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml:83]: Yes

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          EventSchema = "NetworkSession",
+          EventSchemaVersion = "0.2.6",
+          EventResultDetails = "Unknown",
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml]: Done

File: Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_SchemaTest.csv [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+"(2) Info: Missing optional field [DstProcessGuid]"
+"(2) Info: Missing optional field [DstProcessId]"
+"(2) Info: Missing optional field [DstProcessName]"
+"(2) Info: Missing optional field [DstScopeId]"
+"(2) Info: Missing optional field [DstUserId]"
+"(2) Info: Missing optional field [DstUserType]"
+"(2) Info: Missing optional field [DstUsername]"
+"(2) Info: Missing optional field [DstVlanId]"
+"(2) Info: Missing optional field [DstZone]"
+"(2) Info: Missing optional field [DvcAction]"
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_SchemaTest.csv]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                        "MachineType",
+                        agentDetectionInfo_machineType_s,
+                        "OsRevision",
+                        agentDetectionInfo_osRevision_s
+                    )
+      | project-rename
+          EventStartTime = sourceProcessInfo_pidStarttime_t,
+          DstIpAddr = alertInfo_dstIp_s,
+          DvcHostname = agentDetectionInfo_name_s,
+          EventUid = _ResourceId,
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      SentinelOne_CL
+      | where not(disabled) 
+          and event_name_s == "Alerts." 
+          and alertInfo_eventType_s == "TCPV4"
+      | lookup NetworkDirectionLookup on alertInfo_netEventDirection_s
+      | extend 
+          DstPortNumber = toint(alertInfo_dstPort_s),
+          SrcPortNumber = toint(alertInfo_srcPort_s),
+          AdditionalFields = bag_pack(
+                        "MachineType",
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSentinelOne.yaml]: Done

File: Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_SchemaTest.csv [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+"(2) Info: Missing optional field [ThreatFirstReportedTime]"
+"(2) Info: Missing optional field [ThreatId]"
+"(2) Info: Missing optional field [ThreatIpAddr]"
+"(2) Info: Missing optional field [ThreatIsActive]"
+"(2) Info: Missing optional field [ThreatLastReportedTime]"
+"(2) Info: Missing optional field [ThreatName]"
+"(2) Info: Missing optional field [ThreatOriginalConfidence]"
+"(2) Info: Missing optional field [ThreatOriginalRiskLevel]"
+"(2) Info: Missing optional field [ThreatRiskLevel]"
+"(2) Info: Missing recommended alias [Hostname] aliasing non-existent column [DstHostname]"
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_SchemaTest.csv]: Done

File: Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_DataTest.csv [hunk 9]
```diff
@@ -0,0 +1,7 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 11382 records (100.0%) for field [EventProduct] of type [Enumerated]: [""SentinelOne""] (Schema:NetworkSession)"
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_DataTest.csv:2]: pls add SentinelOne as supported value for EventProduct and EventVendor in Tester.csv

File: Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_DataTest.csv [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 11382 records (100.0%) for field [EventVendor] of type [Enumerated]: [""SentinelOne""] (Schema:NetworkSession)"
+"(2) Info: Empty value in 10986 records (96.52%)  in optional field [SrcUsername] (Schema:NetworkSession)"
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_DataTest.csv:4]: SrcIpAddr or DvcIpAddr is all null in sample data:
image

Are we not getting this info much?

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_DataTest.csv (hunk 10 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimNetworkSession/Tests/SentinelOne_ASimNetworkSession_DataTest.csv:5]: Data is already available in raw as well as in csv file with the field name "alertInfo_srcIp" which was mapped with the SrcIpAddr and DvcIpAddr in the Parser.
image

**Comment by niralishah-crest** [comment]
@vakohl we have mapped the inspection fields below in the NetworkSession parser, which we found from SentinelOne Alerts logs.

ThreatConfidence
ThreatOriginalConfidence

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**